client/ayon_core/pipeline/create/product_name.py

```diff
@@ -46,6 +58,15 @@ def get_product_name_template(
         "task_types": task_type
     }
 
+    if is_product_base_type_supported():
```

Comment:
The `is_product_base_type_supported` is to determine if
1. Server does support it.
2. ayon api does support it.

This logic can use the product base type filtering even if none of those are available. So we can add `"product_base_types"` all the time.